pkg/kubelet/images/image_manager_test.go

```diff
@@ -201,12 +231,12 @@ func TestParallelPuller(t *testing.T) {
 
 	useSerializedEnv := false
 	for _, c := range cases {
-		puller, fakeClock, fakeRuntime, container := pullerTestEnv(c, useSerializedEnv)
+		puller, fakeClock, fakeRuntime, container := pullerTestEnv(t, c, useSerializedEnv)
 
 		t.Run(c.testName, func(t *testing.T) {
 			ctx := context.Background()
 			for _, expected := range c.expected {
-				fakeRuntime.CalledFunctions = nil
+				fakeRuntime.CalledFunctions = []string{}
```

Comment:
Why didn't you use fakeRuntime.ClearCalls()

Reply:
I’ll check again but I used that officially and all tests failed. I’ll dig into it more if it’s necessary

Reply:
It is necessary from my point of view. The fact that tests failed could be a sign that they depend on previous test runs and `ClearCalls()` unveiled this.

Reply:
@bart0sh I think this is a valid addition but I think I may need to think more about the implications of this PR.  

I was thinking it would not require a KEP.  

A lot of your suggestions I decided to port over into https://github.com/kubernetes/kubernetes/pull/116231 so we can at least get some of the test additions, error cases and consolidation on image parsing.